app/code/Magento/InventoryCatalog/Test/Integration/UpdateLegacyCatalogInventoryPluginTest.php

```diff
@@ -0,0 +1,151 @@
+        $collectionBeforeChange = $this->oldStockItemRepository->getList($criteria);
+        /** @var StockItemInterface $oldStockItem */
+        $oldStockItem = current($collectionBeforeChange->getItems());
+        $initialQuantity = $oldStockItem->getQty();
+
+        $this->reservationsAppend->execute([
+            $this->reservationBuilder->setStockId(1)->setSku('SKU-1')->setQuantity($reservationQuantity)->build()
+        ]);
+
+        /** @var StockItemCollectionInterface $collectionAfterChange */
+        $collectionAfterChange = $this->oldStockItemRepository->getList($criteria);
+        $oldStockItem = current($collectionAfterChange->getItems());
+        $quantityAfterCheck = $oldStockItem->getQty();
+
+        $this->assertEquals($initialQuantity + $reservationQuantity, $quantityAfterCheck);
+    }
+
+    /**
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/products.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/sources.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stocks.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/source_items.php
+     * @magentoDataFixture ../../../../app/code/Magento/InventoryApi/Test/_files/stock_source_link.php
+     */
+    public function testThatReservationPlacedUpdatesBothOldAndNewStocks()
+    {
+        $this->productRepository = Bootstrap::getObjectManager()->get(ProductRepositoryInterface::class);
+        $reservationQuantity = -5;
+
+        $this->indexer->reindexAll();
+        $this->assertEquals(8.5, $this->getProductQtyInStock->execute('SKU-1', 10));
+
+        /** @var Product $product */
+        $product = $this->productRepository->get('SKU-1');
+
+        /** @var StockItemCriteriaInterface  $criteria */
+        $criteria = $this->stockItemCriteriaFactory->create();
+        $criteria->setProductsFilter([$product->getId()]);
+
+        /** @var StockItemCollectionInterface $collectionBeforeChange */
+        $collectionBeforeChange = $this->oldStockItemRepository->getList($criteria);
+
+        /** @var StockItemInterface $oldStockItem */
+        $oldStockItem = current($collectionBeforeChange->getItems());
+        $initialQuantity = $oldStockItem->getQty();
+        $this->assertEquals(8.5, $initialQuantity);
+
+        $this->reservationsAppend->execute([
+            $this->reservationBuilder->setStockId(10)->setSku('SKU-1')->setQuantity($reservationQuantity)->build()
+        ]);
+
+        /** @var StockItemCollectionInterface $collectionAfterChange */
+        $collectionAfterChange = $this->oldStockItemRepository->getList($criteria);
+        $oldStockItem = current($collectionAfterChange->getItems());
+        $quantityAfterCheck = $oldStockItem->getQty();
+
+        $this->assertEquals($this->getProductQtyInStock->execute('SKU-1', 10), $quantityAfterCheck);
```

Comment:
also, you need to add one more assert that both these value equal to `8.5 - 5`
because your case will work even if Reservation did not affect both of the Inventory data and they stayed the same